pkg/controllers/report/resource/controller.go

```diff
@@ -224,7 +224,7 @@ func (c *controller) updateDynamicWatchers(ctx context.Context) error {
 	}
 	dynamicWatchers := map[schema.GroupVersionResource]*watcher{}
 	for gvk, gvr := range gvrs {
-		logger := logger.WithValues("gvr", gvr, "gvk", gvk)
+		logger := logger.WithValues("gvr", gvr.String(), "gvk", gvk.String())
```

Comment:
Won't `.String()` be called automatically ?

Reply:
It seems that the test result is not. This is also easy to test. Define a structure at random, and then implement String() for the structure. Then print the struct.

Reply:
Well, it's probably implementation specific, I feel like forcing a call to `.String()` is not a good practice, we should leverage the underlying implementation mechanism IMHO.

Reply:
Done. I remove the String() func in other file to make them uniform in style.